server/plugin/virtual_agent.go

```diff
@@ -457,13 +457,21 @@ func (p *Plugin) getPostActionOptions(options []Option) []*model.PostActionOptio
 	return postOptions
 }
 
-func (p *Plugin) CreateMessageAttachment(fileID string) (*MessageAttachment, error) {
+func (p *Plugin) CreateMessageAttachment(fileID, userID string) (*MessageAttachment, error) {
 	var attachment *MessageAttachment
 	fileInfo, appErr := p.API.GetFileInfo(fileID)
 	if appErr != nil {
 		return nil, fmt.Errorf("error getting the file info. Error: %s", appErr.Message)
 	}
 
+	if fileInfo.DeleteAt != 0 {
+		return nil, fmt.Errorf("file is deleted from the server")
+	}
+
+	if fileInfo.CreatorId != userID {
+		return nil, fmt.Errorf("file does not belong to the user: %s", userID)
```

Comment:
this userID is mattermost or servicenow ?.Just a suggestion, We can add that below like for mattermost 
```suggestion
		return nil, fmt.Errorf("file does not belong to the mattermost user: %s", userID)
```